toffy/rosetta.py

```diff
@@ -144,10 +144,16 @@ def clean_rosetta_test_dir(folder_path):
         folder_path (str): base dir for testing, image subdirs will be stored here
     """
 
+    # helper function to prevent linking issues with external drives
+    def ignore_extended_attributes(func, filename, exc_info):
```

Comment:
Is there a more elegant way to handle it? If not, we should add this to alpineer so that we can reuse it in `ark` like mentioned [here](https://github.com/angelolab/ark-analysis/issues/936#issuecomment-1452450703). However, is there not a better way to make sure these hidden files aren't popping up in the first place?

Reply:
@ngreenwald if we go the `dot_clean` or `find` route as suggested in Saeid's answer, I think it may still be worth putting this in `alpineer`. People will be attaching external drives (specifically, NAS drives) at multiple places in the pipeline (including `ark-analysis`).

Reply:
Looking it over, it might be easier to just start with an `os.system` call prior to calling `shutil.rmtree`:

```
os.system("find [path_to_NAS_folder] -type f -name '._*' -delete")
```

This avoid needing to create an explicit tool in `alpineer`. As @ngreenwald mentioned `rmtree` also doesn't pop up as many times as I anticipated, so this should be an easy line to add.

Reply:
I've updated this command to use `subprocess.call` instead.